### PR TITLE
Ignore error if netns path does not exists anymore

### LINKED ISSUE #16: Unable to delete endpoints after reboot
After a reboot, i'm unable to delete some endpoints. Sand will always respond with the following error:

```
An error occured: fail to destroy endpoint: fail to deactivate endpoint: fail to delete endpoint from overlay network: fail to get host namespace handler: no such file or directory
```

### PR DESCRIPTION
**Pull request by EtienneM**
Fix #16

### CODE REVIEW

File: cmd/sand-agent/sand-agent.go [hunk 1]
```diff
@@ -239,8 +239,18 @@ func ensureNetworks(ctx context.Context, c *config.Config, repo network.Reposito
 			MoveVeth:     true,
 		})
 		if err != nil {
-			log.WithError(err).Error("fail to ensure endpoint")
-			continue
+			// if we can't activate the endpoint because the netns path doesn't exist anymore, we
+			// just deactivate it. Otherwise we raise an error.
+			if strings.Contains(err.Error(), "no such file or directory") {
```
Comment (Soulou) [inline cmd/sand-agent/sand-agent.go]: Not the way to check:

https://godoc.org/github.com/pkg/errors#Cause
https://godoc.org/os#IsNotExist

File: network/overlay/delete_endpoint.go [hunk 2]
```diff
@@ -34,7 +38,11 @@ func (m manager) DeleteEndpoint(ctx context.Context, n types.Network, e types.En
 
 	targetfd, err := netns.GetFromPath(e.TargetNetnsPath)
 	if err != nil {
-		return errors.Wrapf(err, "fail to get host namespace handler")
+		err := errors.Wrapf(err, "fail to get host namespace handle from path")
+		if !strings.Contains(err.Error(), "no such file or directory") {
+			return err
+		}
+		log.WithField("endpoint_netns_path", e.TargetNetnsPath).WithError(err).Error("ignore error")
```
Comment (Soulou) [inline network/overlay/delete_endpoint.go]: If we ignore it, is it an error, do we want a notification saying "ignore error" ?

File: network/overlay/ensure_endpoint.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ type overlayEndpoint struct {
 func (m manager) EnsureEndpoint(ctx context.Context, network types.Network, endpoint types.Endpoint, params params.EndpointActivate) (types.Endpoint, error) {
 	hostnsfd, err := netns.Get()
 	if err != nil {
-		return endpoint, errors.Wrapf(err, "fail to get host namespace handler")
+		return endpoint, errors.Wrapf(err, "fail to get current threads network namespace")
```
Comment (Soulou) [inline network/overlay/ensure_endpoint.go]: Current threads? Which threads?

File: cmd/sand-agent/sand-agent.go (hunk 1 above)
Comment (EtienneM) [reply cmd/sand-agent/sand-agent.go]: Thanks I didn't found it when I checked :ok_hand:

File: network/overlay/delete_endpoint.go (hunk 2 above)
Comment (EtienneM) [reply network/overlay/delete_endpoint.go]: No you're right. Lowered to an "info" message

File: network/overlay/ensure_endpoint.go (hunk 3 above)
Comment (Soulou) [reply network/overlay/ensure_endpoint.go]: But I'm really not sure it's clear. I mean I get this message, I don't get it..

**Pull request merged**